pkg/controller/containerizedworkload/containerizedworkload.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package containerizedworkload
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	oamv1alpha2 "github.com/crossplane/crossplane/apis/oam/v1alpha2"
+	"github.com/crossplane/crossplane/apis/workload/v1alpha1"
+
+	"github.com/hasheddan/stack-kubernetes-remote/pkg/reconciler/workload"
+)
+
+const (
+	reconcileTimeout = 1 * time.Minute
+	shortWait        = 30 * time.Second
+	longWait         = 1 * time.Minute
+)
+
+// Reconcile error strings.
+const (
+	errNotContainerizedWorkload = "object is not a containerized workload"
+)
+
+// Reconcile event reasons.
+const (
+	reasonListTraits = "ListedTraits"
+
+	reasonCannotListManualScalerTraits       = "CannotListManualScalerTraits"
+	reasonMultipleManualScalerTraits         = "MultipleManualScalerTraits"
+	reasonCannotPackageContainerizedWorkload = "CannotPackageContainerizedWorkload"
+)
+
+// SetupContainerizedWorkload adds a controller that reconciles ContainerizedWorkloads.
+func SetupContainerizedWorkload(mgr ctrl.Manager, l logging.Logger) error {
+	name := "oam/" + strings.ToLower(oamv1alpha2.ContainerizedWorkloadGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&oamv1alpha2.ContainerizedWorkload{}).
+		Complete(workload.NewReconciler(mgr,
+			workload.Kind(oamv1alpha2.ContainerizedWorkloadGroupVersionKind),
+			workload.WithLogger(l.WithValues("controller", name)),
+			workload.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
+			workload.WithPackager(workload.PackageFn(containerizedWorkloadPackager))))
+}
+
+func containerizedWorkloadPackager(ctx context.Context, app *v1alpha1.KubernetesApplication, obj runtime.Object) error {
```

Comment:
IIRC, there is no service associated with this.  I think we need to decide if we need an additional ingress trait to expose this or create a service for the POC